tests/test_weather_shopper_app.py

```diff
@@ -0,0 +1,248 @@
+                            negative="Total after deletion is incorrect")
+
+        #Checkout to payents page
+        result_flag = checkout(test_mobile_obj)
+        test_mobile_obj.log_result(result_flag,
+                                    positive="Successfully entered checkout page",
+                                    negative="Failed to checkout",
+                                    level="critical")
+
+        #Enter payment details
+        payment_details = conf.valid_payment_details
+        result_flag = test_mobile_obj.select_payment_method(payment_details["card_type"])
+        test_mobile_obj.log_result(result_flag,
+                                positive="Successfully selected payment method",
+                                negative="Failed to select payment method")
+
+        result_flag = test_mobile_obj.enter_email(payment_details["email"])
+        test_mobile_obj.log_result(result_flag,
+                            positive="Successfully entered email",
+                            negative="Failed to enter email")
+
+        result_flag = test_mobile_obj.enter_card_number(payment_details["card_number"])
+        test_mobile_obj.log_result(result_flag,
+                               positive="Successfully entered card number",
+                               negative="Failed to enter card number")
+
+        result_flag = test_mobile_obj.enter_card_expiry(payment_details["card_expiry"])
+        test_mobile_obj.log_result(result_flag,
+                               positive="Successfully entered card expiry",
+                               negative="Failed to enter card expiry")
+
+        result_flag = test_mobile_obj.enter_card_cvv(payment_details["card_cvv"])
+        test_mobile_obj.log_result(result_flag,
+                               positive="Successfully entered card CVV",
+                               negative="Failed to enter card CVV")
+
+        result_flag = test_mobile_obj.submit_payment()
+        test_mobile_obj.log_result(result_flag,
+                               positive="Successfully submitted payment",
+                               negative="Failed to submit payment")
+
+        result_flag = test_mobile_obj.verify_payment_success()
+        test_mobile_obj.log_result(result_flag,
+                               positive="Payment was successful",
+                               negative="Payment was not successful")
+
+
+        # Print out the results.
+        test_mobile_obj.write(f'Script duration: {int(time.time() - start_time)} seconds\n')
+        test_mobile_obj.write_test_summary()
+
+        # Teardown and Assertion.
+        expected_pass = test_mobile_obj.result_counter
+        actual_pass = test_mobile_obj.pass_counter
+
+    except Exception as e:
+        print(f"Exception when trying to run test: {__file__}")
+        print(f"Python says: {str(e)}")
+
+    if expected_pass != actual_pass:
+        raise AssertionError(f"Test failed: {__file__}")
+
```

Comment:
@akkuldn Move all supporting methods to top of test method.

Reply:
Fixed